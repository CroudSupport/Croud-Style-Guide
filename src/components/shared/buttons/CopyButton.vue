<template>
    <button v-if="isSupported" ref="button" :data-clipboard-text="text" :class="buttonClasses" :data-tooltip="tooltipText" data-inverted>
        Copy
    </button>
</template>

<script>
    import Clipboard from 'clipboard'

    /**
     * A button to add text to the clipboard, this button will only show if the browser supports this functionality
     *
     * @see https://clipboardjs.com/
     * @example ./croud-copy-button.md
     */
    export default {
        name: 'croud-copy-button',

        props: {
            /**
             * Text to copy to clipboard
             */
            text: {
                required: true,
                type: String,
            },

            /**
             * Overide button classes
             */
            buttonClasses: {
                type: String,
                default: 'ui basic button copy-button',
            },

            /**
             * Overide tooltip message
             */
            tooltipText: {
                type: String,
                default: 'Copy to clipboard',
            },
        },

        data() {
            return {
                clipboard: null,
            }
        },

        mounted() {
            if (!this.isSupported) return

            this.clipboard = new Clipboard(this.$refs.button)

            this.clipboard.on('success', (e) => {
                this.$emit('success', e)
            })

            this.clipboard.on('error', (e) => {
                this.$emit('error', e)
            })
        },

        computed: {
            isSupported() {
                return Clipboard.isSupported()
            },

        },
    }
</script>
