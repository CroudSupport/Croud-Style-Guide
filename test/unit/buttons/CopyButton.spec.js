import Vue from 'vue'
import CopyButton from '../../../src/components/shared/buttons/CopyButton'

const Constructor = Vue.extend(CopyButton)

const vm = new Constructor({
    propsData: {
        text: 'Copy',
    },
}).$mount()

describe('Copy Button', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
