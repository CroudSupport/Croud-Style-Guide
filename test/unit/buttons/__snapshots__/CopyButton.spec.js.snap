// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy Button should match the snapshot 1`] = `<!--  -->`;
